src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -2771,12 +2771,28 @@ class NodeUtils(object):
                    caller_name())
         # Attempt to take vnodes that match this host offline
         # Assume vnode names resemble self.hostname[#]
+        (vnode_comments, failure) = \
+            fetch_vnode_comments(pbs.event().vnode_list.keys(),
```

Comment:
fetch_vnode_comments() is an expensive operation since it would have to talk to the server. We try to minimize calling this but this maybe the only way to fix this.  Are you okay with the possibility that things could become slow here if connection to the server is hampered?

Reply:
Thank you for the comment. The `take_node_offline()` is called in two places:
 * `_exechost_periodic_handler()`: if remaining orphans > 0
 * if deleting cgroups directory fails

It is not called regularly.

To reduce the traffic, I moved the check of the existence of the `offline_file` at the beginning of the `take_node_offline()`. Now, the `fetch_vnode_comments()` will be called only if the node is not offline. The function `take_node_offline()` will be effectively processed only once per offlining. Sounds better?

Reply:
Great!